spec/requests/notifications_controller_spec.rb

```diff
@@ -228,6 +174,101 @@ def delete_notification(resp_code, matcher)
               read_high_priority.id
             ])
           end
+
+          it "should not bump last seen reviewable in readonly mode" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            Fabricate(:reviewable)
+            Discourse.received_redis_readonly!
+            expect {
+              get "/notifications.json", params: { recent: true }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          ensure
+            Discourse.clear_redis_readonly!
+          end
+
+          it "should not bump last seen reviewable if the user can't seen reviewables" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            Fabricate(:reviewable)
+            expect {
+              get "/notifications.json", params: { recent: true, bump_last_seen_reviewable: true }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          end
+
+          it "should not bump last seen reviewable if the silent param is present" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            Fabricate(:reviewable)
+            expect {
+              get "/notifications.json", params: {
+                recent: true,
+                silent: true,
+                bump_last_seen_reviewable: true
+              }
+              expect(response.status).to eq(200)
+            }.not_to change { user.reload.last_seen_reviewable_id }
+          end
+
+          it "should not bump last seen reviewable if the bump_last_seen_reviewable param is not present" do
+            SiteSetting.enable_experimental_sidebar_hamburger = true
+            user.update!(admin: true)
+            Fabricate(:reviewable)
+            expect {
+              get "/notifications.json", params: { recent: true, silent: true }
```

Comment:
```suggestion
              get "/notifications.json", params: { recent: true }
```

I don't think we should have the `silent` param here because it is the param we use to skip the bump.